cp/cp_config.c

```diff
@@ -223,3 +263,33 @@ fetch_upf_context(uint32_t dpId)
 	/* control should never reach here */
 	return NULL;
 }
+
+struct in_addr
+fetch_dns_primary_ip(uint32_t dpId, bool *present)
+{
+	struct dp_info *dp;
+	struct in_addr dns_p = { .s_addr = 0 };
+	LIST_FOREACH(dp, &appl_config->dpList, dpentries) {
+		if (dpId == dp->dpId) {
+			*present = true;
+			return dp->dns_p;
+		}
+	}
+	*present = get_app_primary_dns(appl_config, &dns_p);
+	return dns_p;
+}
+
+struct in_addr
+fetch_dns_secondary_ip(uint32_t dpId, bool *present)
+{
+	struct dp_info *dp;
+	struct in_addr dns_s = { .s_addr = 0 };
+	LIST_FOREACH(dp, &appl_config->dpList, dpentries) {
+		if (dpId == dp->dpId) {
```

Comment:
added check here to check flags if dns is present and then only return dp level dns.